File: templates/clips/server/lib/debug.ts
/**
 * Verbose-log gating. Enable with `DEBUG=true` (or `DEBUG=1`) — and
 * `CLIPS_DEBUG=true` works too if you want to scope it. Errors and one-time
 * startup messages stay on `console.error`/`console.warn` directly; per-chunk,
 * per-finalize chatter goes through `debugLog` so it stays quiet by default.
 */

const truthy = (v: string | undefined) => v === "1" || v === "true";
const enabled = truthy(process.env.DEBUG) || truthy(process.env.CLIPS_DEBUG);

export function debugLog(...args: unknown[]): void {
  if (enabled) console.log(...args);
}

File: templates/clips/desktop/src-tauri/src/debug.rs
//! Verbose-log gating. Enable with `DEBUG=true` or `CLIPS_DEBUG=true`. Errors
//! and one-time startup messages stay on `eprintln!` directly; per-click /
//! per-frame chatter (toolbar shown, popover blur, capture-excluded applied,
//! …) goes through `dlog!` so it stays quiet by default.

use std::sync::OnceLock;

pub fn debug_enabled() -> bool {
    static ENABLED: OnceLock<bool> = OnceLock::new();
    *ENABLED.get_or_init(|| {
        let truthy = |v: String| matches!(v.as_str(), "1" | "true");
        std::env::var("DEBUG").map(truthy).unwrap_or(false)
            || std::env::var("CLIPS_DEBUG").map(truthy).unwrap_or(false)
    })
}

#[macro_export]
macro_rules! dlog {
    ($($arg:tt)*) => {{
        if $crate::debug::debug_enabled() {
            eprintln!($($arg)*);
        }
    }};
}
